lib/engine/action/place_token.rb

```diff
@@ -11,7 +11,7 @@ def initialize(entity, city:, slot:, token_type: nil)
         @entity = entity
         @city = city
         @slot = slot
-        @token = @entity.find_token_by_type(token_type)
+        @token = @entity.find_token_by_type(token_type) if defined? @entity.find_token_by_type
```

Comment:
i don't see how this code can work. if @token is nil, you're going to hit a null pointer exception. 

i suggest you have company implement find_token_by_type, and given the token type it should return it's corresponding token

Reply:
It worked since the token in the in this action wasn't being used but created inline. This was more hacked together tho so I took your suggestion.